modules/docs/arrow-docs/docs/docs/arrow-core/typeclasses/monadlogic/README.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: docs-core
+title: MonadLogic
+permalink: /docs/arrow/typeclasses/monadlogic/
+redirect_from:
+  - /docs/typeclasses/monadlogic/
+---
+
+## MonadLogic
+
+The `MonadLogic` typeclass provides some functions to control when computations should be performed.
+
+### Main Combinators
+
+#### Kind<F, A>.splitM(): Kind<F, Option<Tuple2<Kind<F, A>, A>>>
+
+Splits a computations into its first result and another computation that computes the rest.
+
+#### <F, A>.interleave(other: Kind<F, A>): Kind<F, A>
+
+Some computations can have an infinite number of potential results. Such computations can cause problems in 
+some circumstances as combined computations may never terminate. `Interleave` allows to combine computations
+in a way that fair consideration of branches is ensured.
```

Comment:
An example using [MonadPlus] here to show unfairness would be good. The paper that the haskell package listed has some good toy examples.